File: TrabajoFinalPrueba1/PegasusWebV1/PegasusWeb/PegasusWeb/Entities/Calificaciones.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PegasusWeb.Entities
{
    public class Calificaciones
    {
        [Key]
        public int Id { get; set; }

        public int? Id_Alumno { get; set; }

        public int? Id_Materia { get; set; }

        public Materia? Materia { get; set; }

        public Usuario? Usuario { get; set; }

        public Byte Calificacion { get; set; }
    }
}

File: pegasusapp/src/models/calificaciones.ts
import type { Materia } from "./materia"
import type { Usuario } from "./usuario"

export interface Calificaciones {
  id: number
  id_Alumno?: number | null
  id_Materia?: number | null
  calificacion: number // Byte en C# se traduce a number en TS (0-255)

  materia?: Materia | null
  usuario?: Usuario | null
}
